TestFoundation/TestThread.swift

```diff
@@ -84,4 +88,44 @@ class TestThread : XCTestCase {
         XCTAssertEqual(thread3.name, "Thread3")
         XCTAssertNotEqual(thread3.name, getPThreadName())
     }
+
+    func test_mainThread() {
+        XCTAssertTrue(Thread.isMainThread)
+        let t = Thread.mainThread
+        XCTAssertTrue(t.isMainThread)
+        let c = Thread.current
+        XCTAssertTrue(c.isMainThread)
+        XCTAssertTrue(c.isExecuting)
+        XCTAssertTrue(c.isEqual(t))
+
+        var started = false
+        let condition = NSCondition()
+        let thread = Thread() {
+            condition.lock()
+            started = true
+            XCTAssertFalse(Thread.isMainThread)
+            XCTAssertFalse(Thread.mainThread == Thread.current)
+            condition.broadcast()
+            condition.unlock()
+        }
+        thread.start()
+
+        condition.lock()
+        if !started {
```

Comment:
I actually just copied the `test_threadStart()` function. You are correct that with a delay starting up the thread, `!started` might be `false` and the `.wait()` will never happen.

There is a `pthread_cond_timedwait()` function to timeout a `.wait()` so I will see if I can find a way to incorporate that to handle the thread never calling `.broadcast()`.

I notice that `NSLock` also has 2 time bounded lock functions that need implementing so I might see if I can implement and incorporate them.